base/channels.jl

```diff
@@ -53,13 +54,82 @@ end
 
 Channel(sz) = Channel{Any}(sz)
 
+# special constructors
+"""
+    Channel(func::Function; ctype=Any, csize=0, taskref=nothing)
+
+Creates a new task from `func`, binds it to a new channel of type
+`ctype` and size `csize`, schedules the task, all in a single call.
+
+`func` must accept the bound channel as its only argument.
+
+If you need a reference to the created task, pass a `Ref{Task}` object via
+keyword argument `taskref`.
+
+Returns a Channel.
+
+```jldoctest
+julia> chnl = Channel(c->foreach(i->put!(c,i), 1:4));
+
+julia> @show typeof(chnl);
+typeof(chnl) = Channel{Any}
+
+julia> for i in chnl
+           @show i
+       end;
+i = 1
+i = 2
+i = 3
+i = 4
+
+```
+
+An example of referencing the created task:
+
+```jldoctest
+julia> taskref = Ref{Task}();
+
+julia> chnl = Channel(c->(@show take!(c)); taskref=taskref);
+
+julia> task = taskref[];
```

Comment:
Julia isn't C, I can't think of any Julia APIs that do this. A separate function for this would make more sense, might not even need to be exported if it isn't used very often. Is there any other way of identifying whether or not a Channel has a task bound to it? Something like being able to ask for `boundtasks(c::Channel)` might be useful if possible